src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTSegmentToExtract.java

```diff
@@ -0,0 +1,169 @@
+// License: GPL. For details, see LICENSE file.
+package org.openstreetmap.josm.plugins.pt_assistant.data;
+
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.RelationMember;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.sort;
+
+/**
+ * Represents a piece of a route that includes all the ways
+ * belonging to the same bundle of itineraries of vehicles
+ * traveling in the same direction
+ *
+ * @author Polyglot
+ *
+ */
+public class PTSegmentToExtract {
+    private final Relation relation;
+
+    private final ArrayList<RelationMember> ptWays;
+    private List<Integer> indices;
+    private List<String> lineIdentifiers;
+    private List<String> colours;
+    private List<String> streetNames;
+
+    /**
+     * Constructor
+     * @param relation  The route or superroute relation for which this route segment is created
+     * ptWays           The list of PTWay members to extract
+     * indices          The indices corresponding to the ways
+     * lineIdentifiers  The ref tag of the route parent route relations of the ways
+     * colours          The colours of the public transport lines of this line bundle
+     */
+
+    public PTSegmentToExtract(Relation relation) {
+        this.relation = relation;
+
+        this.ptWays = new ArrayList<>();
+        this.indices = new ArrayList<>();
+        this.lineIdentifiers = new ArrayList<>();
+        this.colours = new ArrayList<>();
+        this.streetNames = null;
+    }
+
+    /**
+     * Returns the PTWays of this route segment
+     * @return the PTWays of this route segment
+     */
+    public List<RelationMember> getPTWays() {
+        return this.ptWays;
+    }
+
+    /**
+     * Sets the PTWays of this route segment to the given list
+     * @param ptWay           the Way Member to add
+     * @param index           its index in the relation specified in the constructor
+     * @param lineIdentifier  The ref tag of the relation
+     * @param colour          The colour tag of the relation
+     */
+    public void addPTWay(RelationMember ptWay, Integer index, String lineIdentifier, String colour) {
```

Comment:
You can omit the parameters `lineIdentifiers` and `colour`, these are always `relation.get("ref")` and `relation.get("colour")` at the moment.

Reply:
I also realised that I only needed to pass the index. It's obvious from that which Way member was meant.